syntax = "proto3";

package routing;

message Route {
  int64 next_hop = 1;
  double metric = 2;
}

message RouteTable {
  map<uint64, Route> routes = 1;
}

message UpdateMessage {
  uint64 rt_version = 1;
  uint64 source_node = 2;
  map<uint64, Route> updated_routes = 3;
}
